### ISSUE TITLE
UI Realignment

### ISSUE THREAD

**Issue by irmerk** [accordproject/markdown-editor#381]
In the Slate migration, we have strayed a bit from the original editor styling. We need to ensure this is back on to specification, both in `cicero-ui` and in a parent component.

**Comment by irmerk** [comment]
We need to decide what style / design we want in `markdown-editor` (RichTextEditor) _and_ `cicero-ui` (ContractEditor)

**Comment by irmerk** [comment]
I believe the backgrounds of the `FormattingToolbar` icons should be transparent.

**Comment by nik72619c** [comment]
@irmerk is this free to work on ?

**Comment by irmerk** [comment]
Yes, I think it would be most beneficial to propose ideas in this issue first.

**Comment by arteevraina** [comment]
@irmerk I think we should have a `Quick Tour` for the user who is new to markdown editor to get familiar with it.

**Comment by irmerk** [comment]
I think that would be a good separate issue to raise @arteevraina, but also might be better implemented after we get our Storybook settled and see if that solves the issue.

**Comment by nik72619c** [comment]
> Yes, I think it would be most beneficial to propose ideas in this issue first.

Makes sense, will point out some :)

**Comment by shrey-sachdeva2000** [comment]
Hi @irmerk! I designed some mockups for the same. I believe the design should be kept very minimal yet aesthetic, so others could use the component without the hassle to change the default styling.

![mockup_markdown_editor2](https://user-images.githubusercontent.com/47667325/80753415-dc6a0d80-8b4a-11ea-812d-095d7b7e2873.jpg)

Some ideas I think might be good to incorporate are:
1. Add minimal styling to the `markdown editor`.
2. Keep options like Preview and Fullscreen to enhance UX.
3. A simple bar at the bottom of the `markdown editor` that shows the number of lines, characters and words in the editor. (Might also want to keep an option to switch this feature off)

**Comment by Michael-Grover** [comment]
@shrey-sachdeva2000 that looks good but there are some areas with low contrast like here:
![image](https://user-images.githubusercontent.com/40569793/80757727-9db06500-8b02-11ea-87bb-cdad1ea35455.png)

I like the idea of two frames side-by-side for the markdown/preview

I think what we have already serves as a great out of the box style, as everything is already minimal and high contrast. I will share more suggestions later
![image](https://user-images.githubusercontent.com/40569793/80757727-9db06500-8b02-11ea-87bb-cdad1ea35455.png)

**Comment by Michael-Grover** [comment]
I will take a shot at a mockup in the next couple days but as a preview it will probably look the layout of what @shrey-sachdeva2000 sent and the details (such as icons and contrast level) of the existing cicero UI contract editor

**Comment by irmerk** [comment]
> Hi @irmerk! I designed some mockups for the same. I believe the design should be kept very minimal yet aesthetic, so others could use the component without the hassle to change the default styling.

Great! I agree, I think minimal sounds good.

> Some ideas I think might be good to incorporate are:
> 
> 1. Add minimal styling to the `markdown editor` component as well.

Do you mean the component on the left in your screenshot? I think we should focus on this less, because this is the specific demo we use for development. Rather, it should be the last step, I think. This will change in the Storybook implementation (can see locally with [the work](https://github.com/accordproject/web-components) Dan is doing)

> 2. Keep options like Preview and Fullscreen to enhance UX.
> 3. A simple bar at the bottom of the preview that shows the number of lines, characters and words in the editor. (Might also want to keep an option to switch this feature off)

Same point as before, I think these are great for the demo but I'm more specifically focusing on the actual component.

**Comment by shrey-sachdeva2000** [comment]
That makes sense! Thank you for the feedback. @Michael-Grover @irmerk

**Comment by Michael-Grover** [comment]
Can we discuss this at the next tech WG meeting? I was going to bring it up at this week's meeting but the agenda was packed. I will show a mockup at then

**Comment by Michael-Grover** [comment]
As discussed in the Tech WG call, we are going with this design for the default markdown editor and contract editor. 

![image](https://user-images.githubusercontent.com/40569793/81343014-36566000-9082-11ea-943e-9939ab384afa.png)

It is mostly the same as the contract editor demo [as seen in storybook](https://cicero-ui.netlify.app/?path=/story/components-contract-editor--contract-editor) (see below), except:

- lato font for headers
- Formatting bar moved outside the page

![image](https://user-images.githubusercontent.com/40569793/81343263-ad8bf400-9082-11ea-9e06-e8bade5b51a1.png)

When the user scrolls up, the formatting toolbar should stay fixed. Is there a way we can let someone extending this component override the styling for how the formatting toolbar overlaps the editor? Like if someone wants there to be a drop shadow under the toolbar, or a gray line.
![image](https://user-images.githubusercontent.com/40569793/81343608-4884ce00-9083-11ea-9817-8cf6391696be.png)


As for the markdown editor preview, we are transitioning to using a monorepo for UI components, and the markdown preview will become part of Storybook:
https://github.com/accordproject/web-components/issues/1
